File: app/src/main/java/com/dhpcs/liquidity/fragment/AddGameBottomSheetDialogFragment.java
package com.dhpcs.liquidity.fragment;

import android.annotation.SuppressLint;
import android.app.Activity;
import android.os.Bundle;
import android.support.design.widget.BottomSheetDialogFragment;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import com.dhpcs.liquidity.R;

public class AddGameBottomSheetDialogFragment extends BottomSheetDialogFragment {

    public interface Listener {

        void onNewGameClicked();

        void onJoinGameClicked();

    }

    public static final String TAG = "add_game_bottom_sheet_dialog_fragment";

    public static AddGameBottomSheetDialogFragment newInstance() {
        return new AddGameBottomSheetDialogFragment();
    }

    private Listener listener;

    @Override
    public void onAttach(Activity activity) {
        super.onAttach(activity);
        listener = (Listener) activity;
    }

    @Override
    public View onCreateView(LayoutInflater inflater,
                             ViewGroup container,
                             Bundle savedInstanceState) {
        @SuppressLint("InflateParams") View view = getActivity().getLayoutInflater().inflate(
                R.layout.fragment_add_game_bottom_sheet_dialog,
                null
        );

        view.findViewById(R.id.textview_new_game).setOnClickListener(new View.OnClickListener() {

            @Override
            public void onClick(View v) {
                if (listener != null) {
                    listener.onNewGameClicked();
                }
                dismiss();
            }

        });
        view.findViewById(R.id.textview_join_game).setOnClickListener(new View.OnClickListener() {

            @Override
            public void onClick(View v) {
                if (listener != null) {
                    listener.onJoinGameClicked();
                }
                dismiss();
            }

        });

        return view;
    }

    @Override
    public void onDetach() {
        super.onDetach();
        listener = null;
    }

}

File: app/src/main/java/com/dhpcs/liquidity/fragment/AddGameBottomSheetDialogFragment.kt
package com.dhpcs.liquidity.fragment

import android.annotation.SuppressLint
import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.navigation.fragment.findNavController

import com.dhpcs.liquidity.R
import com.google.android.material.bottomsheet.BottomSheetDialogFragment
import kotlinx.android.synthetic.main.fragment_add_game_bottom_sheet_dialog.*

class AddGameBottomSheetDialogFragment : BottomSheetDialogFragment() {

    companion object {

        const val TAG = "add_game_bottom_sheet_dialog_fragment"

        fun newInstance(): AddGameBottomSheetDialogFragment = AddGameBottomSheetDialogFragment()

    }

    @SuppressLint("InflateParams")
    override fun onCreateView(inflater: LayoutInflater,
                              container: ViewGroup?,
                              savedInstanceState: Bundle?): View? {
        return requireActivity().layoutInflater.inflate(
                R.layout.fragment_add_game_bottom_sheet_dialog,
                null
        )
    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        textview_new_game.setOnClickListener {
            CreateGameDialogFragment.newInstance().show(
                    fragmentManager,
                    CreateGameDialogFragment.TAG
            )
            dismiss()
        }
        textview_join_game.setOnClickListener {
            findNavController().navigate(R.id.action_games_fragment_to_join_game_fragment)
            dismiss()
        }
    }

}
